File: SDKs/googleapis.com/cloudscheduler/v1beta1/python/src/sdk/models/shared/pubsubtarget.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json


@dataclass_json
@dataclass
class PubsubTarget:
    attributes: Optional[dict[str, str]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'attributes' }})
    data: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'data' }})
    topic_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'topicName' }})
    

File: SDKs/googleapis.com/cloudscheduler/v1beta1/typescript/src/sdk/models/shared/pubsubtarget.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// PubsubTarget
/** 
 * Pub/Sub target. The job will be delivered by publishing a message to the given Pub/Sub topic.
**/
export class PubsubTarget extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=attributes" })
  attributes?: Map<string, string>;

  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: string;

  @SpeakeasyMetadata({ data: "json, name=topicName" })
  topicName?: string;
}
